tests/hana_test.py

```diff
@@ -343,3 +425,208 @@ def test_sr_cleanup_force(self):
 
         self._hana.sr_cleanup(force=True)
         mock_command.assert_called_once_with('hdbnsutil -sr_cleanup --force')
+
+    def test_get_sr_state_details(self):
+        expected_results = {
+            "Not set (HDB daemon stopped)": { "online": "false", "mode": "none" },
+            "Not set (HDB daemon running)": { "online": "true", "mode": "none" },
+            "Primary (no secondary sync)": {'online': 'true', 'mode': 'primary', 'operation mode': 'primary', 'site id': '1', 'site name': 'NUREMBERG', 'is source system': 'true', 'is secondary/consumer system': 'false', 'has secondaries/consumers attached': 'false', 'is a takeover active': 'false'},
+            "Primary (sync)": {'online': 'true', 'mode': 'primary', 'operation mode': 'primary', 'site id': '1', 'site name': 'NUREMBERG', 'is source system': 'true', 'is secondary/consumer system': 'false', 'has secondaries/consumers attached': 'true', 'is a takeover active': 'false'},
+            "Secondary (sync)": {'online': 'true', 'mode': 'sync', 'operation mode': 'logreplay', 'site id': '2', 'site name': 'PRAGUE', 'is source system': 'false', 'is secondary/consumer system': 'true', 'has secondaries/consumers attached': 'false', 'is a takeover active': 'false', 'active primary site': '1', 'primary masters': 'hana01'},
+            "Primary (kb7023127)": {'online': 'true', 'mode': 'primary', 'site id': '1', 'site name': 'node1'},
+            "Secondary (kb7023127)": {'online': 'true', 'mode': 'sync', 'site id': '2', 'site name': 'node2', 'active primary site': '1'},
+            }
+        for desc, case in _hdbnsutil_sr_state_outputs.items():
+            result = shell.ProcessResult("", 0, case.encode('utf-8'), "".encode('utf-8'))
+            mock_command = mock.Mock()
+            mock_command.return_value = result
+            self._hana._run_hana_command = mock_command
+
+            state = self._hana.get_sr_state_details()
+            mock_command.assert_called_once_with('hdbnsutil -sr_state')
+
+            self.assertEqual(state, expected_results.get(desc, {}))
+
+
+_hdbnsutil_sr_state_outputs = {
```

Comment:
Maybe it would be an option to store all of this data in an external file, in order to have a shorter test file, we could do that in the future though.

Reply:
Yes, agreed :)